pkg/operator/starter.go

```diff
@@ -45,53 +50,90 @@ const (
 	resync = 20 * time.Minute
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	kubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(controllerConfig.KubeConfig, targetName))
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigFile string) error {
+	isHyperShift := guestKubeConfigFile != ""
+	// Kubeconfig received through service account or --kubeconfig arg
+	// is for the management cluster.
+	controlPlaneKubeConfig := controllerConfig.KubeConfig
+	controlPlaneKubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(controlPlaneKubeConfig, targetName))
 	if err != nil {
 		return err
 	}
-	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, "", operatorNamespace, targetNamespace)
+	controlPlaneNamespace := defaultOperandNamespace
+	// Guest kubeconfig is the same as the management cluster one unless guestKubeConfigFile is provided
+	guestKubeClient := controlPlaneKubeClient
+	guestKubeConfig := controlPlaneKubeConfig
+	if guestKubeConfigFile != "" {
+		guestKubeConfig, err = client.GetKubeConfigOrInClusterConfig(guestKubeConfigFile, nil)
+		if err != nil {
+			return fmt.Errorf("Failed to use guest kubeconfig %s: %s", guestKubeConfigFile, err)
+		}
+		controlPlaneNamespace = controllerConfig.OperatorNamespace
+		guestKubeClient = kubernetes.NewForConfigOrDie(rest.AddUserAgent(guestKubeConfig, targetName))
+	}
 
-	configClient, err := configclient.NewForConfig(rest.AddUserAgent(controllerConfig.KubeConfig, targetName))
+	controlPlaneInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(controlPlaneKubeClient, "", controlPlaneNamespace)
+	guestKubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(guestKubeClient, "", defaultOperandNamespace)
```

Comment:
nit: for the sake of readability, I'd use `guestNamespace` instead of `defaultOperandNamespace`.

Reply:
Good idea! Renamed as a new commit.